src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -1210,8 +1211,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
         Public Shared Function ClassifyTupleConversion(source As BoundTupleLiteral, destination As TypeSymbol, binder As Binder, <[In], Out> ByRef useSiteDiagnostics As HashSet(Of DiagnosticInfo)) As ConversionKind
+            If source.Type = destination Then
```

Comment:
Ok, currently IsSameTypeIgnoringCustomModifiers  also ignores tuple field names. I think it would need a flag, because in this case names should not be ignored.